chiselFrontend/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -222,3 +224,11 @@ private[chisel3] object Builder {
     }
   }
 }
+
+/** Allows public access to the naming stack in Builder / DynamicContext.
+  * Necessary because naming macros expand in user code and don't have access into private[chisel3]
+  * objects.
+  */
+object DynamicNamingStack {
+  def apply() = Builder.namingStack
+}
```

Comment:
Macros expand such that this is generated into annotated functions, so it can't be private. I also know of no way to suppress warnings.